src/java.base/share/classes/java/util/concurrent/CopyOnWriteArrayList.java

```diff
@@ -100,6 +100,8 @@ public class CopyOnWriteArrayList<E>
     implements List<E>, RandomAccess, Cloneable, java.io.Serializable {
     private static final long serialVersionUID = 8673264195747942595L;
 
+    private static final Object[] EMPTY_ELEMENTDATA = {};
```

Comment:
Looks like readObject() ought to also be changed to use the EMPTY_ELEMENTDATA if the length of the read array is 0? And perhaps `clone()`.

Reply:
Agreed. You might as well be consistent with this so that no one will need to re-address if some application does zillions of clones of empty lists.

Reply:
I've updated `readObject` to match - thanks!  `clone()` isn't a big win since the array is allocated elsewhere and the application doesn't use that code path, so I'll leave that alone for now.

Reply:
Sorry, I crossed wires with @DougLea 's comment, will add `clone`.

Reply:
`clone()` performs a shallow copy, so there is currently no `Object[]` allocated and therefore nothing to optimize.  I don't think there's any work to do here.

@DougLea @viktorklang-ora can you confirm?

Reply:
:+1: